master/internal/resourcemanagers/resource_pool.go

```diff
@@ -398,11 +373,28 @@ func (rp *ResourcePool) receiveRequestMsg(ctx *actor.Context) error {
 	return nil
 }
 
+func (rp *ResourcePool) fetchAgentStates(ctx *actor.Context) map[*actor.Ref]*agent.AgentState {
+	agents := make([]*actor.Ref, 0, len(rp.agents))
+
+	for k := range rp.agents {
+		agents = append(agents, k)
+	}
+
+	responses := ctx.AskAll(agent.GetAgentState{}, agents...).GetAll()
```

Comment:
can we check the responses for errors so an single agent crash doesn't crash the RP when it does the type assertion below?